Ревизия 1, ФЗ спринта 8

ihonhola opened the pull request:
Добавлена продолжительность и дата старта задач.
Список задач по приоритетам.
Проверка пересечний по времени.
Замена циклов на Stream API.
Новые тесты.

ihonhola committed:
Ревизия 1, ФЗ спринта 8
Добавлена продолжительность и дата старта задач.
Список задач по приоритетам.
Проверка пересечний по времени.
Замена циклов на Stream API.
Новые тесты.

ihonhola committed:
Ревизия 1, ФЗ спринта 8
Добавлена продолжительность и дата старта задач.
Список задач по приоритетам.
Проверка пересечний по времени.
Замена циклов на Stream API.
Новые тесты.

ihonhola committed:
Ревизия 3, ФЗ спринта 8
Добавлена продолжительность и дата старта задач.
Список задач по приоритетам.
Проверка пересечний по времени.
Замена циклов на Stream API.
Новые тесты.
Замечания по стилю

AntonPozdeev commented on src/InMemoryTaskManager.java:
```diff
@@ -1,14 +1,26 @@
+        return time1.compareTo(time2);
+    };
+
+    TreeSet<Task> prioritizedTasks = new TreeSet<>(comparator);
```
private или protected

AntonPozdeev commented on src/InMemoryTaskManager.java:
```diff
@@ -1,14 +1,26 @@
     protected HashMap<Integer, EpicTask> epicTasks;
     protected HashMap<Integer, SubTask> subTasks;
+
+    Comparator<Task> comparator = (task1, task2) -> {
```
private или protected

AntonPozdeev commented on src/InMemoryTaskManager.java:
```diff
@@ -309,4 +371,32 @@ public EpicTask getEpic(int id) {
+    }
+
+    @Override
+    public boolean hasTimeOverlap(Task task1, Task task2) {
```
такой метод не должен быть публичным

AntonPozdeev commented on src/InMemoryTaskManager.java:
```diff
@@ -309,4 +371,32 @@ public EpicTask getEpic(int id) {
+    }
+
+    @Override
+    public boolean isTimeOverlapping(Task task) {
```
такой метод не должен быть публичным

ihonhola committed:
Ревизия 4, ФЗ спринта 8
Добавлена продолжительность и дата старта задач.
Список задач по приоритетам.
Проверка пересечний по времени.
Замена циклов на Stream API.
Новые тесты.
Замечания по стилю.
Поля protected и private, исправлен тест shouldDetectTimeOverlaps
